src/Components/ServiceScene/Breakdowns/PatternsViewTextSearch.tsx

```diff
@@ -0,0 +1,144 @@
+        variables: [new CustomVariable({ name: PATTERNS_TEXT_FILTER, value: state.patternFilter ?? '' })],
+      }),
+    });
+
+    this.addActivationHandler(this.onActivate.bind(this));
+  }
+
+  /**
+   * On click callback to clear current text search
+   */
+  public clearSearch = () => {
+    this.setState({
+      patternFilter: undefined,
+    });
+  };
+
+  /**
+   * Search input onchange callback
+   * @param e
+   */
+  public handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      patternFilter: e.target.value,
+    });
+  };
+
+  /**
+   * Activation handler
+   * @private
+   */
+  private onActivate() {
+    this._subs.add(
+      this.subscribeToState((newState, prevState) => {
+        if (newState.patternFilter !== prevState.patternFilter) {
+          const parent = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+          if (parent.state.patternFrames) {
+            debouncedFuzzySearch(
+              parent.state.patternFrames.map((frame) => frame.pattern),
+              this.state.patternFilter ?? '',
+              this.onSearchResult
+            );
+          }
+        }
+      })
+    );
+  }
+
+  /**
+   * Sets the patterns filtered by string match
+   * @param patterns
+   */
+  setFilteredPatterns(patterns: string[]) {
+    const parent = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+    if (parent.state.patternFrames) {
+      const filteredPatternFrames = parent.state.patternFrames.filter((patternFrame) => {
+        if (this.state.patternFilter && parent.state.patternFrames?.length) {
+          return patterns.find((pattern) => pattern === patternFrame.pattern);
+        }
+        return false;
+      });
+
+      parent.setState({
+        filteredPatterns: filteredPatternFrames,
+      });
+    }
+  }
+
+  /**
+   * Fuzzy search callback
+   * @param data
+   */
+  onSearchResult = (data: string[][]) => {
+    const parent = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+    // If we have a search string
+    if (this.state.patternFilter) {
+      this.setFilteredPatterns(data[0]);
+    } else if (parent.state.filteredPatterns && !this.state.patternFilter) {
+      // Wipe the parent filtered state
+      this.setEmptySearch();
+    }
+  };
+
+  /**
+   * Wipes filtered patterns when search string is empty
+   */
+  private setEmptySearch() {
+    const parent = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+    parent.setState({
+      filteredPatterns: undefined,
+    });
+  }
+}
+
+const styles = {
+  field: css({
+    label: 'field',
+    marginBottom: 0,
+  }),
+  icon: css({
+    cursor: 'pointer',
+  }),
+};
+
+export function PatternTextSearchComponent({ model }: SceneComponentProps<PatternsViewTextSearch>) {
+  const { patternFilter } = model.useState();
+  return (
+    <Field className={styles.field}>
+      <Input
+        suffix={<Icon onClick={model.clearSearch} className={styles.icon} name={'x'} />}
```

Comment:
I like this search input, with icon and suffix, a lot. I'm using something really similar for field breakdowns, and I think, for example, that we should not show the suffix if the input is empty.

I will create a component based on this pattern for reuse in both places.

Reply:
Something like this:

```ts
interface Props extends HTMLProps<HTMLInputElement> {
  onClear(): void;
};

export const SearchInput = ({ value, onChange, placeholder, onClear }: Props) => {
  return (
    <Input
      value={value}
      onChange={onChange}
      suffix={value ? <Icon onClick={onClear} name={'x'} className={styles.clearIcon} /> : undefined}
      prefix={<Icon name="search" />}
      placeholder={placeholder}
    />
  )
};

const styles = {
  clearIcon: css({
    cursor: 'pointer',
  }),
}
```

Reply:
Yes agreed. Should we create a new PR to add the component, or do you want me to add it on this branch?

Reply:
Nah, we can do this afterwards. I'll send the component in another PR.